core/tests/puppeteer-acceptance-tests/user-utilities/voiceover-admin-utils.ts

```diff
@@ -0,0 +1,193 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility functions for voiceover admin page
+ */
+
+import {BaseUser} from '../puppeteer-testing-utilities/puppeteer-utils';
+import testConstants from '../puppeteer-testing-utilities/test-constants';
+import {showMessage} from '../puppeteer-testing-utilities/show-message-utils';
+
+const creatorDashboardPage = testConstants.URLs.CreatorDashboard;
+
+const createExplorationButton = 'button.e2e-test-create-new-exploration-button';
+const dismissWelcomeModalSelector = 'button.e2e-test-dismiss-welcome-modal';
+const textStateEditSelector = 'div.e2e-test-state-edit-content';
+const richTextAreaField = 'div.e2e-test-rte';
+const saveContentButton = 'button.e2e-test-save-state-content';
+const addInteractionButton = 'button.e2e-test-open-add-interaction-modal';
+const interactionEndExplorationInputButton =
+  'div.e2e-test-interaction-tile-EndExploration';
+const saveInteractionButton = 'button.e2e-test-save-interaction';
+const saveChangesButton = 'button.e2e-test-save-changes';
+const saveDraftButton = 'button.e2e-test-save-draft-button';
+
+const publishExplorationButton = 'button.e2e-test-publish-exploration';
+const explorationTitleInput = 'input.e2e-test-exploration-title-input-modal';
+const explorationGoalInput = 'input.e2e-test-exploration-objective-input-modal';
+const explorationCategoryDropdown =
+  'mat-form-field.e2e-test-exploration-category-metadata-modal';
+const saveExplorationChangesButton = 'button.e2e-test-confirm-pre-publication';
+const explorationConfirmPublishButton = 'button.e2e-test-confirm-publish';
+const closeShareModalButton = 'button.e2e-test-share-publish-close';
+
+const explorationSettingsTab = '.e2e-test-settings-tab';
+const editVoiceoverArtistButton = 'span.e2e-test-edit-voice-artist-roles';
+const voiceArtistEditSelector = 'input.e2e-test-new-voice-artist-username';
+const saveVoiceoverArtistEditButton =
+  'button.e2e-test-add-voice-artist-role-button';
+
+const errorToastMessage = 'div.e2e-test-toast-warning-message';
+const closeToastMessageButton = 'button.e2e-test-close-toast-warning';
+
+const updatedVoiceoverArtist = 'div.e2e-test-voiceArtist-role-names';
+const allVoiceoverArtistsList = 'ul.e2e-test-voiceArtist-list';
+
+export class VoiceoverAdmin extends BaseUser {
+  /**
+   * Function to navigate to creator dashboard page
+   */
+  async navigateToCreatorDashboardPage(): Promise<void> {
+    await this.page.goto(creatorDashboardPage);
+  }
+
+  /**
+   * Function to navigate to exploration settings tab
+   */
+  async navigateToExplorationSettingsTab(): Promise<void> {
+    await this.page.waitForSelector(explorationSettingsTab);
```

Comment:
This statement isn't needed since `clickOn` already includes the `page.waitForSelector` statement. You can verify this by checking the method definition of `clickOn` in `puppeteer-testing-utilities/puppeteer-utils.ts`.

Ditto for most instances of this elsewhere.

Reply:
Removed unnecessary instances of `page.waitForSelector`, the remaining ones are to check for the selector with additional prams like `...not([disabled])` `... hidden: true`

Reply:
@hrshkshri I think when `this.clickOn('selector:not([disabled])')` should call `this.page.click('selector:not([disabled])')` and it should still work just fine (you can manually test this). The instances using wait for selector options such as `visible: true` or `hidden: true` can be left alone.